libevm/ExtVMFace.h

```diff
@@ -37,6 +38,46 @@ namespace dev
 namespace eth
 {
 
+/// Reference to a slice of buffer that also owns the buffer.
+///
+/// This is extension to the concept C++ STL library names as array_view
+/// (also known as gsl::span, array_ref, here vector_ref) -- reference to
+/// continuous non-modifiable memory. The extension makes the object also owning
+/// the referenced buffer.
+///
+/// This type is used by VMs to return output coming from RETURN instruction.
+/// To avoid memory copy, a VM returns its whole memory + the information what
+/// part of this memory is actually the output. This simplifies the VM design,
+/// because there are multiple options how the output will be used (can be
+/// ignored, part of it copied, or all of it copied). The decision what to do
+/// with it was moved out of VM interface making VMs "stateless".
+///
+/// The type is movable, but not copyable. Default constructor available.
```

Comment:
It's the opposite -- it's extended `vector_ref`. With simple `bytes` you cannot state e.g. "this is a bytes buffer of 100 bytes, but interesting part is a slice from byte 13 of size 22".

I have in mind another change, to have polymorphic buffer implementation to handle the buffers coming from different VM implementations (like from EVMJIT, where it is allocated by `std::malloc`).